neutron-specs/ip-quota-per-subnet.rst

```diff
@@ -0,0 +1,100 @@
+..
+ This work is licensed under a Creative Commons Attribution 3.0 Unported
+ License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+
+==========================================
+IP quota per Subnet
+==========================================
+
+Neutron doesn't provide a way to specify IP quota per subnet per tenant. This
+is really crucial in environments with many shared subnets (e.g. each rack has
+its own subnet) and administrators want to forbid draining all IPs by one
+tenant.
+
+This spec describes how to extend neutron IPAM to add IP quota per subnet.
+
+
+Problem Description
+===================
+
+Currently Neutron isn't able to limit use of IPs per tenant except:
+
+* a) Fixed-ips quota - Number of fixed IP addresses allowed per tenant. This
+  number must be equal to or greater than the number of allowed instances.
+
+* b) Subnet pools quota - with IPv4, the ``default_quota`` can be set to the
+  number of absolute addresses any given project is allowed to consume from
+  the pool. With IPv6 it is a little different. It is not practical to count
+  individual addresses. To avoid ridiculously large numbers, the quota is
+  expressed in the number of /64 subnets which can be allocated. [1]_
+
+Subnet pool quota is not flexible enough because in environments with many
+subnets, administrators have to create a new subnet pool per subnet to limit
+IPs usage per subnet.
```

Comment:
I think subnet pools would actually work fine for us - we don't need to create a subnet per rack. We could create fewer (but larger) subnets, and then create a subnet pool per rack, and set the quota there. It's equal in work from an operator standpoint, as we have to create something per rack. But, if we use subnet pools, then it should work out of the box instead of needing to patch. Thoughts?

Reply:
I agree, if subnet pools are enough for you we should stick with them since the quota for them is already available. But that's a question to @epim

Reply:
Cool, I talked to @epim about it the other day, he's fine with subnet pools as well. We got this done fast eh? :D

Reply:
Yeah :+1:, can I close this PR then?